fix: add new combined icon component to render double icon assets in one

premiumjibles opened the pull request:
## Description

Currently our side by side asset breaks our components or our alignment semi regularly. 

Discord thread here https://discord.com/channels/554694662431178782/1410133120639959090

E.g.

<img width="438" height="399" alt="image" src="https://github.com/user-attachments/assets/46dfd921-9fe2-4a7b-82d5-61e9feeb25c0" />

<img width="505" height="383" alt="image" src="https://github.com/user-attachments/assets/5c50121e-5534-4490-bd76-bd8b373a505b" />

This PR combined the two into a split single icon.

## Issue (if applicable)

<!-----------------------------------------------------------------------------
If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.
------------------------------------------------------------------------------>

closes #10373

## Risk

Low risk, double icon components are already pretty edge case. This could render our combined icon where maybe we don't want it to?

> High Risk PRs Require 2 approvals

<!-----------------------------------------------------------------------------
Outline the scope of your changes and the risk associated with them. You must use your discretion as an engineer to determine the potential impact of your changes.

WARNING: If your PR introduces a new on-chain transaction or modifies an existing one, please add the "High-Risk" label to this PR and ask for 2 reviewers to approve it before merging.

E.g. an upgrade to `hdwallet` or core state management would be considered higher risk, and might require a full regression test. UI or isolated view changes, or something behind a feature flag may have near zero risk. Small bug fixes might require testing isolated to the specific fix.
------------------------------------------------------------------------------>

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing

* Test that dual icon assets combine in areas where alignment matters (e.g. list scenarios)
* Test that dual icon assets collapse in scenarios with limited space (e.g. asset row)
* Test that other icons are not effected

<!-----------------------------------------------------------------------------
We treat every PR to be merged with the same scrutiny as if we were merging directly to production.

Your PR will not be merged if you do not complete the sections below for our engineering and operations teams to test.
------------------------------------------------------------------------------>

### Engineering

<!-----------------------------------------------------------------------------
Include sufficient information here for an engineer to test your PR. This may include how to test locally, in a built environment, changes to infrastructure etc.
------------------------------------------------------------------------------>

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

<!-----------------------------------------------------------------------------
If your changes have a user-facing impact, describe how a non-technical QA team can functionally test your changes in a preview environment.

If they are not user-facing please describe how to test for any regressions that may occur.
------------------------------------------------------------------------------>

## Screenshots (if applicable)

https://jam.dev/c/fbe70342-3a49-4403-9075-c983621e15a0

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Improved two-asset pair display with a vertical split for clearer identification.

- Refactor
  - Unified icon rendering across transactions, notifications, listings, and menus for more consistent visuals and reduced re-renders.
  - Simplified pairing behavior for single, two, and multi-asset displays.

- Breaking Changes
  - Removed legacy paired-icon configuration affecting some card and icon components' props.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

NeOMakinG commented on src/features/defi/components/PairIcons/PairIcons.tsx:
```diff
@@ -105,10 +109,35 @@ export const PairIcons = ({
   if (!icons?.length) return null
 
+  // Combined mode for exactly 2 icons
+  if (icons.length === 2) {
```
Are we sure we want a fix for only this count? multiple icons will have the same issue than 2 icons as it will show a +X round

<img width="70" height="51" alt="image" src="https://github.com/user-attachments/assets/06bf19a0-b3bc-47dc-b725-c3a419e5de36" />
 
 In this case the layout will also be weird

twblack88 replied:
iIf above 2 we should just show the number icon count. balancer has always done great designing for these use cases. 

We could consider a fancy toggle where if you hit the numbers icon it expands a nested tag with like however many assets are in the pool like this:
<img width="442" height="48" alt="Screenshot 2025-08-28 at 10 59 23 AM" src="https://github.com/user-attachments/assets/e56c9209-0a72-4944-9fde-16c31e52f586" />

tap to expand basically. 

the downside is, of course, doing this on an already expanding row. So we can ticket that experience for the asset itself, since portals keeps throwing these wild multi asset things at us.

premiumjibles commented:
@twblack88 @reallybeard @NeOMakinG To keep things really simple i've just kept the existing behaviour of 3+ icons where it creates the blurry circle. Except i've removed the option to "show first" which makes it side by side (too wide). I've also added an on hover (on click on mobile) to view the assets included.

https://jam.dev/c/9df6843f-b0bc-46da-a4cb-e4a2a0d26211

https://github.com/shapeshift/web/pull/10356/commits/27ec0750acae315a08aa1bbc95917c72d85a5a07

0xApotheosis reviewed (approved):
This is so damn tidy, nice one @premiumjibles.

Tested on responsive mobile, native mobile, and desktop views, all working as expected.

Get in!
